dojo-dungeon/src/starter/DojoStarter.java

```diff
@@ -17,17 +19,59 @@
 import dojo.rooms.Room;
 import dojo.rooms.builder.RoomBuilder;
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.HashMap;
+import java.util.*;
 import java.util.logging.Level;
 
 /** Starter for the dojo-dungeon game. */
 public class DojoStarter {
   private static final String BACKGROUND_MUSIC = "sounds/background.wav";
-  private static final DesignLabel DESIGN_LABEL_FOR_LEVEL_1 = DesignLabel.FOREST;
-  private static final DesignLabel DESIGN_LABEL_FOR_LEVEL_2 = DesignLabel.TEMPLE;
-  private static final DesignLabel DESIGN_LABEL_FOR_LEVEL_3 = DesignLabel.DEFAULT;
-  private static final DesignLabel DESIGN_LABEL_FOR_LEVEL_4 = DesignLabel.FIRE;
+
+  private interface BuildRoomMethod {
+    Room buildRoom(LevelRoom levelRoom, RoomGenerator gen, Room nextRoom, DesignLabel designLabel);
+  }
+
+  private static class LevelRoomLevel {
+    private final DesignLabel designLabel;
+    private final BuildRoomMethod[] buildRoomMethods;
+    private final LevelRoom[] levelRooms;
+
+    private LevelRoomLevel(
+        LevelGraph graph, DesignLabel designLabel, BuildRoomMethod[] buildRoomMethods) {
+      this.designLabel = designLabel;
+      this.buildRoomMethods = buildRoomMethods;
+      levelRooms =
+          Arrays.stream(buildRoomMethods)
+              .map(temp -> new LevelRoom(graph))
+              .toArray(LevelRoom[]::new);
+    }
+
+    /**
+     * Get the level rooms.
+     *
+     * @return the level rooms
+     */
+    public LevelRoom[] getLevelRooms() {
+      return levelRooms;
+    }
+
+    /**
+     * Get the build room methods.
+     *
+     * @return the build room methods
+     */
+    public BuildRoomMethod[] getBuildRoomMethods() {
+      return buildRoomMethods;
+    }
+
+    /**
+     * Get the design label.
+     *
+     * @return the design label
+     */
+    public DesignLabel getDesignLabel() {
+      return designLabel;
+    }
+  }
```

Comment:
Der Starter ist nach meinem Verständnis die Klasse mit der `main`-Methode. D.h. diese Klasse ruft die anderen Klassen auf bzw. instanziiert diese bzw. konfiguriert das `Game`. Es _kann_ also keine Abhängigkeit in der anderen Richtung geben, d.h. es kann per Definition keine andere Klasse geben, die irgendwas aus dem Starter nutzt.

**Warum gibt es hier Getter? Wer soll das von außen aufrufen?**

Reply:
ok, -getter

Reply:
Naja, vor allem: keine `public`-Methoden (außer `main`) ... 

Ich sehe, dass Du die Gettern intern hier in der Klasse verwendest. Da würde ich aber direkt auf die Attribute zugreifen. Nach außen soll (muss) die Implementierung gekapselt sein, aber intern würde ich auf eine weitere Indirektionsstufe eher verzichten. Das bringt aus meiner Sicht keine erkennbaren Vorteile, und kostet durch den jeweiligen Methodenaufruf nochmal extra Zeit (nicht wirklich relevant hier, aber sowas läppert sich auch mal schnell). Und falls Du an der Implementierung (den Attributen) mal was ändern wollen würdest, dann würde das beim Refactoring direkt auffallen bzw. automatisch mitgezogen.

Reply:
ja, du hast recht. rückblick: vorher hatte ich dort einfach eine record verwendet, aber das ist nun nicht mehr möglich, weil die klasse "noch mehr tun muss".

Reply:
> ja, du hast recht. rückblick: vorher hatte ich dort einfach eine record verwendet, aber das ist nun nicht mehr möglich, weil die klasse "noch mehr tun muss".

wenn alle attribute nur einmal im konstruktor gesetzt werden, könnte es immer noch ein record sein - auch in record-klassen kannst du zusätzliche methoden implementieren.

aber für den starter ein record? klingt grad nicht sinnvoll.